gemrb/core/GUI/Control.cpp

```diff
@@ -191,15 +198,13 @@ void Control::BindDictVariable(const varname_t& var, value_t val, ValueRange val
 	SetValue(val);
 	// now that the value range is setup, we can change the dictionary variable
 	VarName = var;
-	
+
 	if (GetDictOp() == BitOp::SET) {
 		// SET implies the dictionary value should always mirror Value
 		UpdateDictValue();
 	} else {
-		value_t dictVal = INVALID_VALUE;
-		if (core->GetDictionary()->Lookup(VarName, dictVal)) {
-			UpdateState(VarName, dictVal);
-		}
+		int value = core->GetVariable(VarName.c_str(), INVALID_VALUE);
+		UpdateState(VarName, value);
```

Comment:
Before, this didn't get called if the dict item didn't exist. It would be fine if UpdateDictValue gets called before, but that doesn't look likely. I suggest we just check for INVALID_VALUE first.